runtime/interpreter/value.go

```diff
@@ -1012,7 +1012,12 @@ func (v *ArrayValue) GetMember(inter *Interpreter, getLocationRange func() Locat
 				v.Append(inter, getLocationRange, invocation.Arguments[0])
 				return VoidValue{}
 			},
-			sema.ArrayAppendFunctionType(inter.ConvertStaticToSemaType(v.Type.ElementType())),
+			func() *sema.FunctionType {
+				arrayStaticType := v.StaticType().(ArrayStaticType)
```

Comment:
Maybe just inline this variable, as it is only used once, and the function wrapper might not be optimized away (just guessing here)

Reply:
Yea, I separate them out for better readability, but makes sense to inline